src/transformers/models/ict/modeling_ict.py

```diff
@@ -0,0 +1,864 @@
+        clusters: Optional[np.ndarray] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> Union[Tuple, BaseModelOutput]:
+        r"""
+        Returns:
+
+        Example:
+        ```python
+        >>> import torch
+        >>> import numpy as np
+        >>> from PIL import Image
+        >>> import requests
+
+        >>> from transformers import AutoImageProcessor, IctModel
+
+        >>> image_processor = image_AutoImageProcessor.from_pretrained("sheonhan/ict-imagenet-256")
+        >>> model = IctModel.from_pretrained("sheonhan/ict-imagenet-256")
+
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+        >>> pixel_values = image_processor(image, return_tensors="pt").pixel_values
+        >>> clusters = image_processor.clusters
+
+        >>> # create random boolean mask of shape (batch_size, num_patches)
+        >>> bool_masked_pos = torch.randint(low=0, high=2, size=(pixel_values.shape[0] * pixel_values.shape[1])).bool()
+
+        >>> outputs = model(pixel_values, bool_masked_pos=bool_masked_pos, clusters=clusters)
+        ```"""
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+        clusters = clusters if clusters is not None else self.config.clusters
+
+        if pixel_values is None:
+            raise ValueError("You have to specify pixel_values")
+
+        if clusters is None:
+            raise ValueError("You have to specify clusters")
+
+        outputs = self.transformer(
+            pixel_values,
+            bool_masked_pos=bool_masked_pos,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+        logits = outputs[1]
+        batch_size, sequence_length, _ = logits.shape
+        height = width = math.floor(sequence_length**0.5)
+
+        # modified from https://github.com/raywzy/ICT/blob/59dd12d374d47cdf0dce90923017ca3657e6aa0b/Transformer/inference.py#L107-L126
+        # clusters = np.rint(127.5 * (clusters + 1.0))
+        clusters = torch.from_numpy(clusters)
+        original_images = [clusters[pixel_values[i]].view(height, width, 3) for i in range(batch_size)]
+        recovered_pixel_values = self.sample_mask(
+            pixel_values, logits, bool_masked_pos, temperature=self.config.temperature, top_k=self.config.top_k
+        )
+        recovered_images = [clusters[recovered_pixel_values[i]].view(height, width, 3) for i in range(batch_size)]
+
+        reshaped_bool_masked_pos = [
+            bool_masked_pos.reshape(height, width) if bool_masked_pos is not None
+            # Handle without boolean mask
+            else torch.full((height, width), 1)
+            for _ in range(batch_size)
+        ]
```

Comment:
This doesn't match the docstring. Is `bool_masked_pos` of shape `(batch_size, height * width)` or `(height * width)`? 


If `(batch_size, height * width)` then we can reshape directly
```
        if bool_masked_pos is None: 
           reshaped_bool_masked_pos = torch.full((batch_size, height, width), 1)
       else: 
           reshaped_bool_masked_pos = bool_masked_pos.reshape(batch_size, height, width)
```

Or it's size `(height, width)`, in which case there's no need to do a for loop:

```
        if bool_masked_pos is None: 
           reshaped_bool_masked_pos = torch.full((batch_size, height, width), 1)
       else: 
           reshaped_bool_masked_pos = torch.tile(bool_masked_pos, (batch_size, 1, 1))
```